### PR TITLE
Use Babel instead of system locale to format dates

### PR DESCRIPTION
**Pull request by Kwpolska**
* Fixes issues #2606 and #3121.
* Also fixes the locale situation for users who have incomplete locales or misconfigured systems.
* Supersedes the `babeldates` plugin for v7.
* Adds a few extra nifty features with date formatting (but not much else).
* Removes a lot of ugly and dead code.

I’m not a fan of how the threading support currently looks in LocaleBorg, but without it, everything goes awry.  Also, our translators may need to fix the date strings if (a) my guesses were wrong or (b) the CLDR data isn’t perfect in the context (cf. Polish translation of `Posts for {month_year}`.)

### CODE REVIEW

File: tests/README.rst [hunk 1]
```diff
@@ -1,106 +0,0 @@
-.. title: The Nikola Test Suite
-.. slug: tests
-.. date: 2012/03/30 23:00
-
-The Nikola Test Suite
```
Comment (felixfontein) [inline tests/README.rst]: Is there a reason why you removed the tests README.rst completely?

Comment (felixfontein) [reply tests/README.rst]: Ok, I see: "Remove terribly outdated test documentation"
Makes sense :)

Comment (Kwpolska) [reply tests/README.rst]: The document is fairly outdated, and after removing the now-irrelevant parts, it wouldn’t really be useful for anything.

**Comment by Kwpolska** [comment]
@felixfontein Babel supports basically all locales spoken by humans and some of the obscure ones, too. [Data from CLDR is used in many products](http://cldr.unicode.org/#TOC-Who-uses-CLDR-), and some customizations are doable in the locale itself. I don’t think supporting overrides is worth it, especially at the price of added complexity (what if the plugin needs thread-safety, like `locale` did?)

**Comment by felixfontein** [comment]
@Kwpolska Unfortunately there are languages/dialects where CLDR is not sufficient, or where no official spelling exists and spelling can vary by individual persons. An example is Swiss German (locale gsw). I checked the CLDR month names, they are mostly OK, [except the one for October](https://unicode.org/cldr/charts/latest/summary/gsw.html#1344), which contains two "o"s, and I don't like that and don't want to have that in my blog :) Currently I'm using a custom date formatter for that (since Python doesn't know about gsw anyway), but if that is removed I'll have to rework my themes to stop using Nikola's date formatting facilities, and also have to fork the archives plugin to be able to control the archive names.

I understand the argument about complexity, but I would shove that burden to the plugin. If the plugin really needs thread safety, it has to do that by itself.

**Pull request merged**